### PR TITLE
Added TSV loss function and tests

### PR DESCRIPTION
**Pull request by trq5014**

### CODE REVIEW

File: src/mpol/losses.py [hunk 1]
```diff
@@ -267,3 +267,30 @@ def PSD(qs, psd, l):
...
 
     return loss
+
+
+def TSV(sky_cube):
+    r"""
+    Calculate the total square variation (TSV) loss in the image dimension (R.A. and DEC). Following the definition in `EHT-IV 2019 <https://ui.adsabs.harvard.edu/abs/2019ApJ...875L...4E/abstract>`_ Promotes the image to be edge smoothed which may be a better reoresentation of the truth image `K. Kuramochi et al 2018 <https://ui.adsabs.harvard.edu/abs/2018ApJ...858...56K/abstract>`_.
+
+    Args:
+        sky_cube (any 3D tensor): the image cube array :math:`I_{lmv}`, where :math:`l` is R.A., :math:`m` is DEC, and :math:`v` is the channel (velocity or frequency) dimension. Should be in sky format representation.
```
Comment (iancze) [inline src/mpol/losses.py:277]: This should be more specific about how the sky cube is packed. Specifically, frequency in ndim=1, m in ndim=2, and l in ndim=3.

File: test/losses_test.py [hunk 2]
```diff
@@ -96,3 +96,23 @@ def test_entropy_cube():
 
     cube = torch.ones((nchan, npix, npix), dtype=torch.float64)
     losses.entropy(cube, 0.01)
+
+
+def test_tsv():
+    cube = torch.rand((1, 3, 3))
```
Comment (iancze) [inline test/losses_test.py:102]: rather than hard coding the 3 in the array creation and the 2 in the loops, how about an npix=3 and loop over (npix - 1), making it easy to extend this to larger npix values?

File: test/losses_test.py [hunk 3]
```diff
@@ -96,3 +96,23 @@ def test_entropy_cube():
 
     cube = torch.ones((nchan, npix, npix), dtype=torch.float64)
     losses.entropy(cube, 0.01)
+
+
+def test_tsv():
```
Comment (iancze) [inline test/losses_test.py:101]: it would be good to add some comments to this test describing what's being done

File: test/losses_test.py [hunk 4]
```diff
@@ -96,3 +96,23 @@ def test_entropy_cube():
...
+    tsv_val = losses.TSV(cube)
+    for_val = 0
+    for i in range(2):
+        for j in range(2):
+            for_val += (cube[:, i+1, j] - cube[:, i, j]) ** 2 + (cube[:, i, j+1] - cube[:, i, j]) ** 2
+    assert tsv_val == for_val
+
+
+def test_tv_image():
+    cube = torch.rand((1, 3, 3))
```
Comment (iancze) [inline test/losses_test.py:112]: similar comments as for test_tsv

**Review by iancze — COMMENTED** [review]
This looks great. I have some minor comments for clarity and documentation.

File: src/mpol/losses.py [hunk 5]
```diff
@@ -103,7 +103,7 @@ def TV_image(sky_cube, epsilon=1e-10):
     Calculate the total variation (TV) loss in the image dimension (R.A. and DEC). Following the definition in `EHT-IV 2019 <https://ui.adsabs.harvard.edu/abs/2019ApJ...875L...4E/abstract>`_ Promotes the image to be piecewise smooth and the gradient of the image to be sparse.
 
     Args:
-        sky_cube (any 3D tensor): the image cube array :math:`I_{lmv}`, where :math:`l` is R.A., :math:`m` is DEC, and :math:`v` is the channel (velocity or frequency) dimension. Should be in sky format representation.
+        sky_cube (any 3D tensor): the image cube array :math:`I_{lmv}`, where :math:`l` is R.A. in :math:`ndim=1`, :math:`m` is DEC in :math:`ndim=2`, and :math:`v` is the channel (velocity or frequency) dimension in :math:`ndim=1`. Should be in sky format representation.
```
Comment (iancze) [inline src/mpol/losses.py:106]: I think R.A. is in ndim=3?

Comment (trq5014) [reply src/mpol/losses.py:106]: Whoops that was a typo. I'll fix that now

**Comment by iancze** [comment]
Great, thanks!
